File: SCH/SCH.Client/src/app/sch/student/interfaces/student-grid-request.ts
import { GridRequest } from '../../../interfaces/grid-request';

export interface StudentGridRequest extends GridRequest {
  firstNameValue1?: string | null;
  firstNameOperator1?: string | null;
  firstNameValue2?: string | null;
  firstNameOperator2?: string | null;
  firstNameFilterConcatOperator?: string | null;

  lastNameValue1?: string | null;
  lastNameOperator1?: string | null;
  lastNameValue2?: string | null;
  lastNameOperator2?: string | null;
  lastNameFilterConcatOperator?: string | null;

  emailValue1?: string | null;
  emailOperator1?: string | null;
  emailValue2?: string | null;
  emailOperator2?: string | null;
  emailFilterConcatOperator?: string | null;

  phoneNumberValue1?: string | null;
  phoneNumberOperator1?: string | null;
  phoneNumberValue2?: string | null;
  phoneNumberOperator2?: string | null;
  phoneNumberFilterConcatOperator?: string | null;

  ssnValue1?: string | null;
  ssnOperator1?: string | null;
  ssnValue2?: string | null;
  ssnOperator2?: string | null;
  ssnFilterConcatOperator?: string | null;

  startDateOperator1?: string | null;
  startDateValue1?: string | null;
  startDateValue2?: string | null;
  startDateFilterConcatOperator?: string | null;
  startDateOperator2?: string | null;
  startDateValue3?: string | null;
  startDateValue4?: string | null;

  isActive?: boolean | null;
}

File: SCH/SCH.Models/Students/ClientDtos/StudentGridRequest.cs
namespace SCH.Models.Students.ClientDtos
{
    using SCH.Models.Common.GridEntities;

    public class StudentGridRequest : GridRequest
    {
        public string? FirstNameValue1 { get; set; }
        public string? FirstNameOperator1 { get; set; }
        public string? FirstNameValue2 { get; set; }
        public string? FirstNameOperator2 { get; set; }
        public string? FirstNameFilterConcatOperator { get; set; }

        public string? LastNameValue1 { get; set; }
        public string? LastNameOperator1 { get; set; }
        public string? LastNameValue2 { get; set; }
        public string? LastNameOperator2 { get; set; }
        public string? LastNameFilterConcatOperator { get; set; }

        public string? EmailValue1 { get; set; }
        public string? EmailOperator1 { get; set; }
        public string? EmailValue2 { get; set; }
        public string? EmailOperator2 { get; set; }
        public string? EmailFilterConcatOperator { get; set; }

        public string? PhoneNumberValue1 { get; set; }
        public string? PhoneNumberOperator1 { get; set; }
        public string? PhoneNumberValue2 { get; set; }
        public string? PhoneNumberOperator2 { get; set; }
        public string? PhoneNumberFilterConcatOperator { get; set; }

        public string? SSNValue1 { get; set; }
        public string? SSNOperator1 { get; set; }
        public string? SSNValue2 { get; set; }
        public string? SSNOperator2 { get; set; }
        public string? SSNFilterConcatOperator { get; set; }

        public string? StartDateOperator1 { get; set; }
        public string? StartDateValue1 { get; set; }
        public string? StartDateValue2 { get; set; }
        public string? StartDateFilterConcatOperator { get; set; }
        public string? StartDateOperator2 { get; set; }
        public string? StartDateValue3 { get; set; }
        public string? StartDateValue4 { get; set; }

        public bool? IsActive { get; set; }
    }
}
